src/Tools/ExternalAccess/NativeAOT/Internal/InterceptorUtils.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+internal static class InterceptorUtils
+{
+    /// <summary>
+    /// If the call denoted by <paramref name="invocation"/> is intercepted in <paramref name="compilation"/>, returns the method symbol for the interceptor.
+    /// Otherwise, returns null.
+    /// </summary>
+    /// <remarks>
+    /// Calling this method forces completion of attributes on all source symbols.
+    /// If using this API in an analyzer, you may want to only access it through
+    /// the callback to 'RegisterCompilationEndAction', which will avoid potential negative perf impact in the IDE.
+    /// </remarks>
+    internal static MethodSymbol? GetInterceptor(InvocationExpressionSyntax invocation, CSharpCompilation compilation, CancellationToken cancellationToken)
```

Comment:
Yeah, here's the sample I wrote for the api review mail.

```cs
public static void AnalyzeCall(IInvocationOperation invocation)
{
    // native AOT analyzer needs to perform its checks based on the "interceptor" method, not the original method.
    if (invocation is { Syntax: InvocationExpressionSyntax invocationExpression, SemanticModel.Compilation: CSharpCompilation csCompilation })
    {
        // use a possible EA API to solve it?
        var interceptor = InterceptorUtils.GetInterceptor(invocationExpression, csCompilation, cancellationToken: default);
    }
}
```

My thinking was, if the consumer wasn't IOperation-based but still wanted this info, it would still be possible to get the interceptor this way.

Reply:
Is resolving the InvocationExpressionSyntax via `IInvocationOperation.Syntax` sufficient here?